client/deposits/index.tsx

```diff
@@ -10,12 +10,37 @@ import React from 'react';
  */
 import Page from 'components/page';
 import { TestModeNotice } from 'components/test-mode-notice';
+import BannerNotice from 'components/banner-notice';
+import DepositSchedule from 'components/deposits-overview/deposit-schedule';
+import { useAllDepositsOverviews } from 'data';
 import DepositsList from './list';
 
+const NextDepositNotice: React.FC = () => {
+	const {
+		overviews: { account },
+	} = useAllDepositsOverviews();
+
+	const depositSettings = wcpaySettings.accountStatus?.deposits;
+	const isDepositsUnrestricted =
+		wcpaySettings.accountStatus.deposits?.restrictions ===
+		'deposits_unrestricted';
+
+	if ( ! depositSettings || ! isDepositsUnrestricted ) {
```

Comment:
When `account` is still being fetched, reference to `account.deposits_schedule` will throw an error.

I got this error on page load.

![Screenshot 2023-12-13 at 00 16 57](https://github.com/Automattic/woocommerce-payments/assets/73803630/1a0da1c1-2bbc-43f0-b19f-e33d20e745d0)

I think we can fix by returning early here.

```suggestion
	if ( ! depositSettings || ! isDepositsUnrestricted || ! account ) {
```

Reply:
I'm getting this error when testing on my site, too.

Reply:
Doh, thanks!

Reply:
TS not helping me out here! I'll fix up the return type of `useAllDepositsOverviews()` in a followup PR.

Reply:
Resolved in 8ec00b158c23403c74a7b624ce5793d284ce4d64